apps/ledger-live-mobile/e2e/specs/firmwareUpdate.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { scrollToText } from "../helpers";
+import DeviceSelectionPage from "../models/settings/deviceSelectionPage";
+
+let portfolioPage: PortfolioPage;
+let settingsPage: SettingsPage;
+let debugSettingsPage: DebugSettingsPage;
+let deviceSelectionPage: DeviceSelectionPage;
+
+const updatableDeviceInfo = {
+  bootloaderVersion: "1.16",
+  hardwareVersion: 0,
+  hasDevFirmware: false,
+  isBootloader: false,
+  isOSU: false,
+  isRecoveryMode: false,
+  languageId: 0,
+  majMin: "2.1",
+  managerAllowed: false,
+  mcuBlVersion: undefined,
+  mcuTargetId: undefined,
+  mcuVersion: "2.30",
+  onboarded: true,
+  pinValidated: true,
+  providerName: null,
+  seTargetId: 855638020,
+  seVersion: "2.0.0",
+  targetId: 855638020,
+  version: "2.0.0",
+};
+
+describe("Receive BTC", () => {
+  beforeAll(async () => {
+    await loadConfig("1AccountBTC1AccountETH", true);
+    portfolioPage = new PortfolioPage();
+    settingsPage = new SettingsPage();
+    debugSettingsPage = new DebugSettingsPage();
+    deviceSelectionPage = new DeviceSelectionPage();
+  });
+
+  it("should go to Settings", async () => {
+    await portfolioPage.navigateToSettings();
+  });
+
+  it("should navigate to Debug settings", async () => {
+    await scrollToText("Debug", "SettingsScrollView");
+    await settingsPage.navigateToDebugSettings();
+  });
+
+  it("should navigate to Firmware update flow", async () => {
+    bridge.mockDeviceAction("updateFirmware");
+
+    await debugSettingsPage.tapFeaturesDebugFlow();
+    await debugSettingsPage.tapFirmwareUpdateFlow();
+  });
+
+  it("should start update", async () => {
+    await deviceSelectionPage.tapAddWithBluetooth();
+    await deviceSelectionPage.tapConnectAnExistingLedger();
+
+    bridge.addDevices();
+
+    await waitFor(deviceSelectionPage.getNanoXByName("David"))
+      .toExist()
+      .withTimeout(3000);
+
+    await deviceSelectionPage.tapNanoXByName("David");
+
+    await waitFor(deviceSelectionPage.getPairingWithNanoXText("David"))
+      .toExist()
+      .withTimeout(10000);
+
+    bridge.open();
+
+    await waitFor(debugSettingsPage.getTriggerUpdateButton())
+      .toExist()
+      .withTimeout(10000);
+
+    await debugSettingsPage.tapTriggerUpdate();
+  });
+
+  it("should correctly render all states of the update", async () => {
+    bridge.updateDeviceActionState({
+      deviceAction: "updateFirmware",
+      newState: {
+        lockedDevice: false,
+        progress: 0.5,
+        step: "installingOsu",
+        error: null,
+      },
+    });
+
+    await expect(debugSettingsPage.getUpdateProgress()).toHaveText("0.5");
+    await expect(debugSettingsPage.getUpdateLockedDevice()).toHaveText("false");
+    await expect(debugSettingsPage.getUpdateStep()).toHaveText("installingOsu");
+    await expect(debugSettingsPage.getUpdateError()).toHaveText("null");
+
+    await new Promise((resolve) => setTimeout(() => resolve("a"), 10000));
```

Comment:
What's the purpose of this promise?

Reply:
This is a left-over. I'm removing it, it was there only to give me a pause after the test was over so I can see what was going on.